doc/write-integration-test.md

```diff
@@ -1,3 +1,132 @@
 # Wasm Extension Integration Test Dev Guide
+---
 
-TODO
+Before deploying your Wasm extensions in production, it is **highly** recommended to write integration tests which load and run the extension with Envoy binary. To make writing such integration test easier, the integration test framework from istio/proxy repo will be used, which is the same framework used to test Istio telemetry extensions.
+
+## Overview
+---
+
+The reference document about the test framework could be found [here](https://godoc.org/github.com/istio/proxy/test/envoye2e/driver). At the high level, the test framework:
+* Downloads Envoy binary, which is built by Istio proxy postsubmit and used in istio proxy docker container.
+* Spawns up a xDS server locally, which serves customizable xDS resources. In your test, the XDS resource will point to your own Wasm extension.
+* Spawns up Envoy processes locally with customizable bootstrap template.
+* Executes test logic which sends request through Envoy, then examines extension logic accordingly.
+
+To take a more detailed look, the test framework models every operation in the test as a [`Step`](https://godoc.org/github.com/istio/proxy/test/envoye2e/driver#Step). For example, one of the most important steps is [`Envoy`](https://godoc.org/github.com/istio/proxy/test/envoye2e/driver#Envoy), which downloads and spawns an Envoy process in `Run` method, and stops the process in `Clean up` method.
+
+Every `Run` method implementation takes a [`Params`](https://godoc.org/github.com/istio/proxy/test/envoye2e/driver#Params) struct as input, which should be created at the beginning of the test and carries test context that is shared by all `Steps`. Information provided by `Params` including:
+* Port assignment, such as port assigned to client Envoy, server Envoy, and XDS server.
```

Comment:
_port numbers assigned to the client Envoy, the server Envoy, and the xDS server_